apps/src/sites/studio/pages/levels/contract_match.jsx

```diff
@@ -182,50 +172,69 @@ $(window).load(function () {
         </div>
       );
     }
-  });
+  }
 
-  var DomainsList = React.createClass({
-    propTypes: {
+  class DomainsList extends React.Component {
+    static propTypes = {
       domainTypes: PropTypes.array.isRequired,
       onDomainAdd: PropTypes.func.isRequired,
-    },
-    render: function () {
-      var self = this;
-      var sortedDomains = this.props.domainTypes.sort(function (a,b) {
-        return a.order > b.order;
-      });
-      var typeChoiceNodes = sortedDomains.map(function (object) {
-        return (
-          <div className="clear" key={object.key}>
-            <TypeChooser
-              order={object.order}
-              type={object.type}
-              key={object.key}
-              onTypeChange={curry(self.props.onDomainChange, object.key)}
-            />
-            <button className="btn domain-x-button" onClick={curry(self.props.onDomainRemove, object.key)}>Remove</button>
-          </div>
-        );
-      });
+      onDomainChange: PropTypes.func.isRequired,
+      onDomainRemove: PropTypes.func.isRequired,
+    };
+
+    render() {
+      const sortedDomains = this.props.domainTypes.sort((a,b) => (a.order > b.order));
+      const typeChoiceNodes = sortedDomains.map(object => (
+        <div className="clear" key={object.key}>
+          <TypeChooser
+            order={object.order}
+            type={object.type}
+            key={object.key}
+            onTypeChange={(...args) => this.props.onDomainChange(object.key, ...args)}
+          />
+          <button
+            className="btn domain-x-button"
+            onClick={(...args) => this.props.onDomainRemove(object.key, ...args)}
+          >
+            Remove
+          </button>
+        </div>
+      ));
       return (
         <div className="domainsList">
           {typeChoiceNodes}
           <button className="btn domain-add-button" onClick={this.props.onDomainAdd}>Add Domain</button>
         </div>
       );
     }
-  });
+  }
 
-  var TypeChooser = React.createClass({
-    propTypes: {
+  class TypeChooser extends React.Component {
+    static propTypes = {
       onTypeChange: PropTypes.func.isRequired,
       type: PropTypes.string,
-    },
-    selectmenuChange: function (selectChange) {
+    };
+
+    selectmenuChange = (selectChange) => {
       this.props.onTypeChange(selectChange.target.value);
-    },
-    render: function () {
-      var divStyle = {
+    };
+
+    componentDidMount() {
+      $(ReactDOM.findDOMNode(this)).coloriconselectmenu({
+        select: function () {
```

Comment:
nit: should we replace with arrow function?

Reply:
Actually, it's important that we don't.  The existing function uses `this` but is _not_ bound, manually or by React, because it expects `this` to be the `<select>` DOM element, not the React component (see the signature of `addSquareIconToButton`).